azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,360 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+from datetime import datetime
+import glob
+from heapq import heappush, heappop
+import os
+import tarfile
+import zipfile
+
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, rm_files, append_file, append_items_to_file
+from azurelinuxagent.common.utils.shellutil import run_command
+from azurelinuxagent.common.utils.textutil import safe_shlex_split
+
+
+LOG_COLLECTOR_DIR = '/var/lib/waagent/logcollector'
+TRUNCATED_FILES_DIR = '/var/truncated'
+
+OUTPUT_RESULTS_FILE_PATH = os.path.join(LOG_COLLECTOR_DIR, "results.txt")
+OUTPUT_ARCHIVE_PATH = os.path.join(LOG_COLLECTOR_DIR, "logs.tar")
+COMPRESSED_ARCHIVE_PATH = os.path.join(LOG_COLLECTOR_DIR, "logs.zip")
+
+MUST_COLLECT_FILES = [
```

Comment:
These files will be given priority if the total size of files to be collected exceeds `ARCHIVE_SIZE_LIMIT`. The higher the file is in this list, the higher its priority. For example, `/var/log/waagent*` files will be collected before `/var/lib/waagent/GoalState.*.xml' files.